### PR TITLE
fix(frontend): mark the base amount, not the derived leg, on order errors

### PR DESCRIPTION
**Pull request by artkorotkikh-dfinity**
# Motivation

An order below the pair's minimum notional reddened the **You get at least** input:

> Order value must be at least 1 ckUSDT

That row is the derived leg — `disabled` on both sides, a readout of `base × price`. Marking it points the user at a field they cannot act on, and says nothing about what to change. The same held for a buy whose cost exceeded the free quote balance, which reddened **You pay at most**.

The only inputs a user can adjust are the base amount and the price.

# Changes

Every amount error now marks the base amount, the one input that can clear any of them:

| Error | Was | Now |
| --- | --- | --- |
| `balance` (sell) | base | base |
| `balance` (buy) | **quote** | base |
| `lot` | base | base |
| `min_notional` / `max_notional` | **quote** | base |

The messages travel with the mark rather than staying next to the figure they are denominated in, so the mark and the explanation agree — a red message under an unmarked field, or a marked field with no message, reads worse than either. The copy holds up: *"Costs 1200 ckUSDC, only 1000 free"* sits under the amount you would reduce, and *"Order value must be at least 1 ckUSDT"* under the amount you would raise.

The price is the other lever for a notional error, but raising it to clear a min-notional pushes a resting order further from the book — a worse fix than adjusting size — so the amount is the one worth pointing at.

Drops the now-dead quote-side plumbing: `onQuoteCustomValidate`, `quoteAmountError`, and its message line.

# Tests

`validateAmount` returns at most one `errorKind`, so exactly one field is ever marked — the edge cases are about which one, and each is covered:

- Two existing specs asserted the old placement (buy balance and min-notional under the quote box); both now assert the base box, with the negative assertion kept so a regression cannot pass silently.
- Added a spec that the derived quote input stays unmarked while a quote-denominated error fires, asserting `toBeDisabled()` alongside it so the reason is recorded next to the expectation.
- Sell balance and lot-multiple specs already asserted the base box and are unchanged.
- No amount error can fire before an amount is entered (`validateAmount` returns early on `!(baseAmount > 0)`) or before a price is set for the notional bounds, so neither produces an unexplained mark.
- The generic parse error still takes precedence over the amount error under the base row.

`npm run lint -- --max-warnings 0`, `npm run check` (0 errors / 0 warnings) and the `components/trading` suite (219 tests) pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code) v2.1.202
Model: Claude Opus 5 (`claude-opus-5`)

### CODE REVIEW

File: src/frontend/src/tests/lib/components/trading/limit-order/LimitOrderTradePairBox.svelte.spec.ts [hunk 1]
```diff
@@ -159,8 +162,24 @@ describe('LimitOrderTradePairBox', () => {
...
+		const [baseInput, quoteInput] = getAllByTestId(TOKEN_INPUT_CURRENCY_TOKEN);
+
+		expect(quoteInput).toBeDisabled();
+		expect(quoteInput.closest('div.py-2')).not.toHaveTextContent(
+			en.trading.limit_order.error_min_notional.split(' $')[0]
+		);
+		expect(baseInput.closest('div.py-2')).toHaveTextContent(
+			en.trading.limit_order.error_min_notional.split(' $')[0]
+		);
 	});
```
[the remark this answers is not in the document]
Comment (artkorotkikh-dfinity) [reply src/frontend/src/tests/lib/components/trading/limit-order/LimitOrderTradePairBox.svelte.spec.ts]: Right on both counts — fixed in 1046355.

The spec now asserts the mark itself, via the class `TokenInputCurrency` toggles. There is no semantic equivalent to assert instead: nothing in this tree sets `aria-invalid`, so the class is the only observable for a state that is purely visual — which is the whole subject of this PR.

Chasing the first point turned up a second, worse problem that your comment did not mention but led straight to. I reintroduced the bug to check the new assertion actually bites, and **the original spec passed with the quote row marked** — so it was not merely incomplete, it was inert. Both rows run their own 300ms validation debounce in `TokenInputContent`. Waiting on the base row's mark resolves as soon as *its* debounce lands, while the quote row's is still pending, so the negative assertion read an unmarked row every time regardless of the code under test. Fake timers advancing past both leaves nothing in flight for either row to be judged on.

Verified in both directions this time:

- quote row re-marked → `× marks the base amount and leaves the derived quote input unmarked`
- fix in place → 13 passed

The `split(' $')` is gone too; the expectation is built with `replacePlaceholders` like the sibling specs.

🤖 Reply from [Claude Code](https://claude.com/claude-code) v2.1.202 — Model: Claude Opus 5 (`claude-opus-5`)

**Comment by artkorotkikh-dfinity** [comment]
I decided not to touch it for now as the solution is not elegant enough still. Will get back to this sometime later
